infrastructure/containers/postgres/automatic-backup/backup_or_restore.py

```diff
@@ -75,6 +113,13 @@ def backup_database(db_backup_gzip_fpath):
     )
     run_shell_command(backup_cmd, env_vars={"PGPASSWORD": os.environ["POSTGRES_PASSWORD"]})
 
+    # upload the backup to S3
+    print("Backing up the database as", object_name, "to S3")
+    upload_backup_to_s3(create_s3_client(), db_backup_gzip_fpath, BACKUP_BUCKET, object_name)
```

Comment:
Hah, nice! Can you do the `create_s3_client()` call the line above so you can store it in a variable and declare the type in place?

`s3_client: BaseClient = create_s3_client()`

Reply:
I'm pretty sure the python version for this is too old to parse statically typed variables. I was having issues with fstrings as well which is why I assume i'm getting parsing errors. Should I look into running a more recent version of python?

Reply:
Leaving this for later. Need to run a new version of python to be able to statically type variables. apt-get seems to only want to run python 3.5 no matter what you do. I also tried building a python image and installing postgres inside of that instead of the other way around and didn't have any luck. I think this should be a separate PR